### PR TITLE
Clarify documentation and indicate that rect_clip_content affects only CanvasItem based nodes

### LINKED ISSUE #37683: rect_clip_content of Control node does not clip children in all cases
**Godot version:** 3.2.1

**OS/device including version:** Windows 10

**Issue description:**
In some cases children and their children get displayed outside of the rect of the parent, even though _rect_clip_content_ is set to _true_ on the parent.

According to the documentation of the control class https://docs.godotengine.org/en/stable/classes/class_control.html#class-control-property-rect-clip-content I would expect, that all children get clipped within the bounds of the clipping parent.

This happens in the following setup:

```
- Control
  - Node
    - TileMap
```

Thank you for having a look at this issue.

**Steps to reproduce:**
Create a scene according to the described setup.
Add Tiles to the TileMap that are outside of the rect of the Control node.
Set rect_clip_content on the node Control to true.
The issue is visible in the editor and when the scene is run.

**Workaround:**
Change the type of Node to Control or Node2D.

**Alternative solution:**
Change the documentation of rect_clip_content and describe this behavior.

**Minimal reproduction project:**
[ClipTileMapBug.zip](https://github.com/godotengine/godot/files/4449448/ClipTileMapBug.zip)

### PR DESCRIPTION
**Pull request by Sauermann**
Resolves #37683

### CODE REVIEW

File: doc/classes/Control.xml [hunk 1]
```diff
@@ -838,7 +838,7 @@
 			Controls whether the control will be able to receive mouse button input events through [method _gui_input] and how these events should be handled. Also controls whether the control can receive the [signal mouse_entered], and [signal mouse_exited] signals. See the constants to learn what each does.
 		</member>
 		<member name="rect_clip_content" type="bool" setter="set_clip_contents" getter="is_clipping_contents" default="false">
-			Enables whether rendering of children should be clipped to this control's rectangle. If [code]true[/code], parts of a child which would be visibly outside of this control's rectangle will not be rendered.
+			Enables whether rendering of CanvasItem based children should be clipped to this control's rectangle. If [code]true[/code], parts of a child which would be visibly outside of this control's rectangle will not be rendered.
```
Comment (clayjohn) [inline doc/classes/Control.xml:841]: You can provide a link to CanvasItem here just by putting square brackets around it.
```suggestion
			Enables whether rendering of [CanvasItem] based children should be clipped to this control's rectangle. If [code]true[/code], parts of a child which would be visibly outside of this control's rectangle will not be rendered.
```

**Comment by clayjohn** [comment]
Thanks! :)

Now you will have to squash the commits together so that there is just one commit. explained [here](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase)

**Comment by Sauermann** [comment]
@clayjohn thank you for walking me through these steps! please let me know if there is anything else to do.

**Review by clayjohn — APPROVED** [review]

**Comment by clayjohn** [comment]
Looks good! Congratulations on your first contributor! Let's hope there are many more to come.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**